Composer/packages/client/src/pages/botProject/adapters/ABSChannels.tsx

```diff
@@ -247,7 +248,37 @@ export const ABSChannels: React.FC<RuntimeSettingsProps> = (props) => {
       }
       await httpClient.put(url, data, { headers: { Authorization: `Bearer ${token}` } });
       if (channelId === CHANNELS.TEAMS) {
-        setShowTeamsCallOut(true);
+        const createResults = await httpClient.get(url, { headers: { Authorization: `Bearer ${token}` } });
+        if (!createResults.data?.properties?.properties?.acceptedTerms === true) {
+          if (
+            await OpenConfirmModal(formatMessage('Microsoft Teams terms and conditions'), null, {
+              disabled: true,
+              confirmText: formatMessage('Agree'),
+              checkboxProps: {
+                kind: 'doubleConfirm',
+                checkboxLabel: (
+                  <div>
+                    {formatMessage('I agree to the ')}
```

Comment:
use formatMessage.rich, you can find many examples in the code by searching that term.

Reply:
ah thanks for the tip! I knew there must be a way